scipy/interpolate/rbf.py

```diff
@@ -235,6 +269,31 @@ def __call__(self, *args):
         if not all([x.shape == y.shape for x in args for y in args]):
             raise ValueError("Array lengths must be equal")
         shp = args[0].shape
-        xa = np.asarray([a.flatten() for a in args], dtype=np.float_)
+        xa = np.asarray([a.flatten() for a in args], dtype=np.float64)
         r = self._call_norm(xa, self.xi)
-        return np.dot(self._function(r), self.nodes).reshape(shp)
+        n_points, n_dims = shp[0] if len(shp) else 1, len(args)
+        POLY = np.empty((n_points, self.n_features), dtype=np.float64)
+        deg = self.degree if self.degree else -1
+        combs = combinations(n_dims, deg)
+        for i, c in enumerate(combs):
+            POLY[:, i] = xa[c, :].prod(axis=0)
+        return np.dot(np.hstack((self._function(r), POLY)), self.nodes).reshape(shp)
+
+    def _generate_augmented_matrix(self, xi):
+        """
+        Generates the augmented matrix as described in Section 2.1 of:
+        http://www.sciencedirect.com/science/article/pii/S0021999116301632
+        by adding polynomial coefficients generated from the data points.
+        """
+        n_points, n_dims = xi.shape
+        # closed-form formula for number of polynomial coefficients
+        deg = self.degree if self.degree else -1
+        self.n_features = int(comb(deg + n_dims, n_dims))
```

Comment:
`features` are the generated polynomial coefficients. Updated the variable name and description.